contracts/USDCPool.sol

```diff
@@ -12,163 +12,45 @@ contract USDCPool is
-        require(lockedAmount >= amount, "Pool Error: You are trying to unlock more funds than have been locked for your contract. Please lower the amount.");
-        lockedAmount = lockedAmount.sub(amount);
-    }
-
-    /*
-     * @nonce calls by CruizePutOptions to lock the premiums
-     * @param amount Amount of premiums that should be locked
-     */
-    function sendPremium() external  payable onlyOwner {
-        lockedPremium = lockedPremium.add(msg.value);
-    }
-
-    /*
-     * @nonce calls by CruizePutOptions to unlock the premiums after an option's expiraton
-     * @param amount Amount of premiums that should be unlocked
-     */
-    function unlockPremium(uint256 amount) external override onlyOwner {
-        require(lockedPremium >= amount, "Pool Error: You are trying to unlock more premiums than have been locked for the contract. Please lower the amount.");
-        lockedPremium = lockedPremium.sub(amount);
-    }
-
-    /*
-     * @nonce calls by CruizeCallOptions to send funds to liquidity providers after an option's expiration
-     * @param to Provider
-     * @param amount Funds that should be sent
-     */
-    function send(address payable to, uint256 amount)
-        external
-      override
-        onlyOwner
-        
-    {
-        require(to != address(0));
-        require(lockedAmount >= amount, "Pool Error: You are trying to unlock more premiums than have been locked for the contract. Please lower the amount.");
-        to.transfer(amount);
-    }
-
-    /*
-     * @nonce Returns provider's share in USDC
-     * @param account Provider's address
-     * @return Provider's share in USDC
-     */
-    function shareOf(address account) external view returns (uint256 share) {
-        if (totalSupply() > 0)
-            share = totalBalance().mul(balanceOf(account)).div(totalSupply());
-        else
-            share = 0;
-    }
-
-    /*
-     * @nonce Returns the amount of USDC available for withdrawals
-     * @return balance Unlocked amount
-     */
-    function availableBalance() public view returns (uint256 balance) {
-        return totalBalance().sub(lockedAmount);
-    }
-
-    /*
-     * @nonce Returns the total balance of USDC provided to the pool
-     * @return balance Pool balance
-     */
-    function totalBalance() public override view returns (uint256 balance) {
-        return address(this).balance.sub(lockedPremium);
-    }
 
-    function _beforeTokenTransfer(address from, address, uint256) internal view override {
-        require(
-            lastProvideTimestamp[from].add(lockupPeriod) <= block.timestamp,
-            "Pool: Withdrawal is locked up"
-        );
+    function withdraw(uint256 WUSDC_Count) external returns (uint256) {
+        require(WUSDC_Count <= balanceOf(msg.sender), "Pool: Amount is too large");
+        require(WUSDC_Count > 0, "Pool: Amount is too small");
+        _burn(msg.sender, WUSDC_Count* 10**uint(decimals()));
```

Comment:
`We don't handle decimals in the smart contract this should be handled by frontend side, because its not a good practice to handle decimals in the smart contract`

Reply:
ok, thanks will change this and push the code.